logger.go

```diff
@@ -64,104 +61,121 @@ func (l *Logger) WithPrefix(prefix string) *Logger {
 // WithData creates a Logger from an existing logger, using the specifed data.
 // Any existing data on the input Logger will be replaced.
 func (l *Logger) WithData(vars ...interface{}) *Logger {
-	if l == nil {
-		return nil
-	}
 	return &Logger{
 		loggingT: l.loggingT,
 		data:     vars,
 		prefix:   l.prefix,
 	}
 }
 
+// AppendData creates a Logger from an existing logger,
+// appending the provided data to the data in the existing logger.
+func (l *Logger) AppendData(vars ...interface{}) *Logger {
+	newData := make([]interface{}, len(l.data))
```

Comment:
We know the new capacity here: len(l.data) + len(vars); we may as well initialize with it.